### PR TITLE
Change sub-page navigation to MD Tabs-like component

### LINKED ISSUE #4214: Migrate to material design nav-tabs
### Observed behavior

Current material design tabs are used in a nonstandard way. 
Here are some ways in which their placement can be a concern:
- Visibility of the tabs is lost as the user scrolls down the page. As a result, the information hierarchy is difficult to process and may be lost in some instances.
- Tabs take up a lot of vertical space

### Expected design

Nav-tabs being a standard component working with the app-bar. See material design specifications for more detail: https://material.io/design/components/tabs.html#scrollable-tabs. 

Had discussions with Christian and Jessica about whether we should use material tabs with icons or without. It's a consensus that using the nav-tabs with icons is a good upgrade, and, alternatively, removing the icons completely wouldn't be too detrimental, as there would also be some benefits to that as well (less noisy, more screen space, moving nav-tabs into the app-bar draws attention to tabs even without icons). But, let's try with icons first!

navtabs

### Scroll behavior
Appbar hides on scroll-down
Appbar reappears on scroll-up 

### Mobile behaviors
From material design: offset the first tab on the left to signal that tabs are scrollable.
Example:
screen shot 2018-09-10 at 11 18 56 am

### Nest point counter within profile dropdown
Notes:
- Point counter is NOT a hyperlink and does not link to the profile tab
- Purely informative static text

screen shot 2018-09-04 at 10 49 06 am

### PR DESCRIPTION
**Pull request by jonboiser**
### Summary

1. Re-styles `KNavBar` and `KNavBarLink` to look like a Material Design Tabs bar
1. Removes unused `type` prop from `KNavBarLink`
1. Moves Learner Points Icon to the CoreMenu dropdown.

**Tabs on Plugins**

screen shot 2018-10-23 at 10 40 31 am

**Learner Points on Core Menu**

screen shot 2018-10-23 at 10 40 53 am

### Reviewer guidance

1. Test behavior of the top navigation element when going between subpages
1. "" When going between pages with a simple single-level "immersive toolbar" and the two-level navigation.
1. "" When attempting to view a page as an unauthorized user
1. "" In various screen sizes

### References

Fixes #4214 
…

----

### Contributor Checklist

- [x] Contributor has fully tested the PR manually
- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [x] If there are any front-end changes, before/after screenshots are included
- [x] If this is an important user-facing change, PR or related issue has a 'changelog' label

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] Reviewer has fully tested the PR manually
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Link to diff of internal dependency change is included
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] Contributor is in AUTHORS.rst

### CODE REVIEW

**Comment by christianmemije** [comment]
image

File: kolibri/core/assets/src/views/KNavbar/index.vue [hunk 1]
```diff
@@ -49,13 +56,20 @@
...
         return this.enoughSpace ? this.elementWidth : this.elementWidth - 38 * 2;
       },
+      wrapperClass() {
+        if (!this.enoughSpace) {
+          return ['wrapper-narrow'];
+        }
+      },
     },
     mounted() {
       this.checkSpace();
```
Comment (christianmemije) [inline kolibri/core/assets/src/views/KNavbar/index.vue:75]: Need to account for the 16px padding in `availableWidth` inside `checkSpace()`

**Review by christianmemije — APPROVED** [review]
Looks great! :1st_place_medal:

**Comment by indirectlylit** [comment]
> * [x]  PR has the correct target branch and milestone

not possible because the milestone didn't exist yet ;-)

**Pull request merged**